baseとなるHTMLの作成

daichan2019 opened the pull request:
## issue
fix #1 

## 対応ブランチ
feature/make-base

daichan2019 committed:
[Add] baseの現在作成した状態をpush

daichan2019 committed:
[Change] refs #1

画像ファイルの命名を英語に変更、それに伴うhtml, cssのパスの変更

daichan2019 committed:
[Add] refs #1

高校生ページを追加

daichan2019 committed:
[Add] refs #1

お知らせページを追加

daichan2019 committed:
[Add] refs #1

コミットもれファイルのpush(p-news.scss, p-pagination.scss)

daichan2019 committed:
[Add] refs #1

看護師ページの追加

daichan2019 committed:
[Add] refs #1

お問い合わせページの追加

daichan2019 committed:
[Add] refs #1

お知らせ記事ページの追加

daichan2019 committed:
[Fix] refs #1

class名変更、component化（p-platform__text-wrap, c-box--shadow）

daichan2019 committed:
[Remove] refs #1

oldディレクトリの削除

daichan2019 committed:
[Fix] refs #1

c-buttonの追加で、ボタンのcomponent化

daichan2019 commented:
d9f836d

.c-buttonでボタンをcomponentにしました。

daichan2019 committed:
[Fix] refs #1

Top pageのデザイン変更に伴うテキストや画像のサイズ変更
box-shadowのcomponent化

daichan2019 committed:
[Fix] refs #1
トップページ以外のデザイン変更箇所の変更

daichan2019 commented:
b1f33d8
2730446

カンプのテキストが更新されていたので修正しました。

daichan2019 committed:
[Fix] refs #1

歯科医師ページをカンプで更新された通りにテキストを変更

daichan2019 committed:
[Fix] refs #1

jQueryの設定、テキストの修正

daichan2019 committed:
[Fix] refs #1

トップページのカンプの変更に合わせてテキストを修正

daichan2019 committed:
[Fix] refs #1

奨学金ページの幅調整

daichan2019 committed:
[Fix] refs #1

iframeタグのhtml上にスタイル直書きを修正

daichan2019 committed:
[FIx] refs #1

スクロールアニメーションの実装

daichan2019 committed:
[Fix] refs #1

全てのページにトップページへ戻るアイコンを追加

daichan2019 committed:
[Fix] refs #1

validatorで確認後のError対応

daichan2019 committed:
[Fix] refs #1

paginationの矢印をfont awesomeへ変更

daichan2019 committed:
[Add] refs #1

install autoprefixer

daichan2019 committed:
[Fix] refs #1

ベンダープレフィックスの

daichan2019 committed:
[Fix] refs #1

box-shadowの付け忘れ発見したので追加

daichan2019 committed:
[Fix] refs #1

c-buttonの付け忘れ修正

daichan2019 commented:
再度、カンプのテキストの差し替えや細かいところの見直しをしました。

daichan2019 committed:
[Fix] refs #1

jsのconstが「,」だったものをconstで定義するように修正

daichan2019 committed:
[Fix] refs #1

ヘッターグロナビの背景色に色がついているものは、「医学生向け奨学金制度」のみでお願いします（現在はホバー時に色が付くようになったいる)→修正

daichan2019 committed:
[Fix] refs #1

フッター内のメニュー→デザインのとおりに修正

daichan2019 commented:
f345a18
71a685b


・ヘッターグロナビの背景色に色がついているものは、「医学生向け奨学金制度」のみでお願いします（現在はホバー時に色が付くようになったいる)
・フッター内のメニュー

上記２点をご指摘のとおり修正しました。

daichan2019 committed:
[Fix] refs #1

footerのLINEアイコンを追加
TOPページのサイドバーにLINEアイコンを追加

daichan2019 committed:
[Fix] refs #1

「事業所のご案内」一部テキスト→修正完了

daichan2019 committed:
[Fix] refs #1

「奨学金をお考えのあなたへ」下の画像・テキスト変更→修正
医学生向け奨学金制度のご案内各種テキスト→修正

daichan2019 committed:
[Fix] refs #1

「病院見学・実習」の詳しく見るボタンにリンク先を入れています→修正
企画内の画像・テキスト→修正

daichan2019 committed:
[Fix] refs #1

看護師ページのvideoの高さ調整

daichan2019 committed:
[Fix] refs #1

奨学金制度」内の詳しく見るボタンにリンクがついています→修正

daichan2019 committed:
[Fix] refs #1

「高校生向け企画」内の画像テキスト変更→修正
「奨学金制度はこちは」のボタンにリンクの設定をお願いします→修正

daichan2019 commented:
d02130a
a163b12
5079076
029d5c5
076cc7b
406c716
c7d895f
現状川村さんからいただいたコメントとfigmaとのデザインの修正を全て行いました。

daichan2019 commented:
川村さんからいただいているコメントに対しての質問がありましたのでまとめて貼ります。
```
QAリスト
1.TOPページサイドバーのLINEのアイコンにQRコードのリンクをつける場合、そのリンクURLが欲しい。
2.■医学生向け奨学金制度
お申込用紙・規定→現状は既定のpdfのリンクを貼っている。リンクは申し込み用紙と規定2つともに飛ばす必要があるのか、
3.■歯科医・歯科学生
就業条件内のテキストにリンクがついています→リンク先わからない
4.■高校生
「奨学金制度はこちは」のボタンにリンクの設定をお願いします→「医学生向け奨学金制度」のページへ遷移の認識で良いか
```

daichan2019 committed:
[Fix] refs #1

■歯科医・歯科学生
 就業条件内のテキストにリンクがついています→修正

daichan2019 commented:
dd73a6d
■歯科医・歯科学生
 就業条件内のテキストにリンクがついています
→本番と同じリンクをはり修正しました

daichan2019 committed:
[Fix] refs #1

■医学生向け奨学金制度
・お申込用紙・規定→ボタンクリック時のリンクを2つに飛ばすように修正

daichan2019 committed:
[Fix] refs #1

リンクの飛び先修正

daichan2019 committed:
[Fix] refs #1

看護師ページのテキスト差し替え
高校生ページのリンク修正

daichan2019 committed:
[Fix] refs #1

各ページのfirst-viewのサイズを360pxに修正

daichan2019 commented:
bf03956
■医学生向け奨学金制度
・お申込用紙・規定→ボタンクリック時のリンクを2つに飛ばすように修正

daichan2019 commented:
1cfec45
7d05f3c 
564f5d1
figmaで再度確認した際に発見した修正漏れをfixしました。

namitarox commented on wp-content/base/index.html:
```diff
@@ -0,0 +1,715 @@
+          </h1>
+          <nav class="l-header-bottom__nav">
+            <ul class="l-header-bottom__nav-list">
+              <li class="l-header-bottom__nav-item">
```
ここ、右側に線いらん

daichan2019 replied:
3855f98

いろいろ試行錯誤したのですが、.is-active時の右の縦線は消すことができません。

「医学生向け奨学金制度」以外のページに.is-acitveが乗った場合も横の線はどうしても乗ってしまいます。
そこまでクリティカルな修正でなければそのままということにはいかないでしょうか？？

namitarox replied:
そもそも`hover`したときにこのデザインにするわけじゃないから、`is-active`ではないかな。
となると、先頭のメニューだけクラス名変えれば縦線消せると思うけど。

```
l-header-bottom__nav-link{
  &_primary{
    専用のデザイン
  }

  &_secondary +  &_secondary{
     border-left: 1px solid #999;
  }
{

```
多分こんな感じ

namitarox replied:
そもそも
```
.l-header-bottom__nav-item + .l-header-bottom__nav-item {
    border-left: 1px solid #999;
}
```
のコードが見当たらんねんけど、どこに書いてる？

namitarox replied:
**html**
```
<ul class="l-header-bottom__nav-list">
              <li class="l-header-bottom__nav-item primary">
                <a class="l-header-bottom__nav-link" href=""
                  >医学生向け<br />奨学金制度</a
                >
              </li>
              <li class="l-header-bottom__nav-item">
                <a class="l-header-bottom__nav-link" href="">医師・医学生</a>
              </li>
              <li class="l-header-bottom__nav-item">
                <a class="l-header-bottom__nav-link" href=""
                  >歯科医・歯科学生</a
                >
              </li>
              <li class="l-header-bottom__nav-item">
                <a class="l-header-bottom__nav-link" href="">看護学生</a>
              </li>
              <li class="l-header-bottom__nav-item">
                <a class="l-header-bottom__nav-link" href=""
                  >医師を目指す高校生</a
                >
              </li>
            </ul>
```

**CSS**
```
.l-header-bottom__nav-link {
  align-items: center;
  -webkit-box-align: center;
  -webkit-box-pack: center;
  color: #20b97b;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -ms-flex-align: center;
  -ms-flex-pack: center;
  height: 35px;
  justify-content: center;
  position: relative;
  width: 160px;
}
.primary .l-header-bottom__nav-link {
  background-color: #20b97b;
  color: #ffffff;
  position: relative;
  border-radius: 5px;
  height: 60px;
}

.l-header-bottom__nav-item:not(.primary) + .l-header-bottom__nav-item:not(.primary) {
  border-left: solid 1px #999999;
}
```

雑やけど実現はできた。
まずデフォルトのメニューの形が角丸になってる必要がない。
角丸である必要がないから、`::before`もいらん
背景に色があるものは高さを60px
背景に色がないものは35pxをそれぞれで指定すれば、縦線の高さも制御できる。

bemで書いてると、`.primary .l-header-bottom__nav-link `だけprimaryで始まってて、コードの見え方汚いから、ここはどうするべきか考えて。
そもそもこうなることがあるからsmaccsの方が組み変えやすいし保守性高いねんけどな。

namitarox replied:
一旦修正のサンプルコード
https://github.com/namitarox/mm-project-_shimane-min/pull/3

daichan2019 replied:
参考にいたします。
ありがとうございます。
.is-activerとしたのは、そのページをクリックしたら（＝そのページに遷移したら）そのページのnavだけ緑色に活性化していると判断したからです。hover時の名残で.is-activeを残していた訳ではないです。jsで制御するなら.js-プレフィックスつけてやるのもありかなと思いました。
FLOCSSでやると確かにSCSSの部分冗長になってしまうのは次反省します。。。

namitarox replied:
>.is-activerとしたのは、そのページをクリックしたら（＝そのページに遷移したら）そのページのnavだけ緑色に活性化している

これ確認します。
デザイン見た感じやと、常に「医学生向け奨学金制」のみmodifyやと思ってんけど。

daichan2019 replied:
adfadcc

> デザイン見た感じやと、常に「医学生向け奨学金制」のみmodifyやと思ってんけど。

LINEでのやりとりで「医学生向け奨学金制」だけ背景色変えることわかったので、いただいたサンプルコードをもとにデザイン通り実現しました。

activeのところだけmodifierつけて`l-header-bottom__nav-link--primary`に変更。
borderの付け方は
```
&__nav-item:nth-of-type(n + 3) {
    border-left: 1px solid $grey;
  }
```
でnav-itemの3つ目以降のleftにつけるとデザイン再現できるので、コードの見通し的にわかりやすいと考え、このようにリファクタしました。

namitarox commented on wp-content/base/index.html:
```diff
@@ -0,0 +1,715 @@
+        <img src="./images/01_top_page/scroll-up.png" alt="ページトップへ" />
+      </a>
+    </div>
+    <footer class="l-footer">
```
フッター全体的にidを入れて、各ページの該当箇所にスクロールが移動するようにする。

daichan2019 replied:
ed1a5ed
フッターのアイテムに紐づくsectionにidをふり、フッターのアイテムにiidのhrefを対応させました。
スムーススクロールはjsで制御するように実装

namitarox replied:
これ、wp化した後変更する必要ありげ

daichan2019 replied:
例えば、トップページから「医師・医学生」の「学生サポートセンターの紹介」sectionにも飛べるようにする
という感じにすれば良いですかね？

namitarox replied:
そやな！

daichan2019 replied:
bca9b71

wp化した際にカスタムnavでいじる必要ありますが、現状のHTML作成時点でも他のページの該当リンクへ飛べるようにはしておきました。

namitarox reviewed (commented):
一旦ここまで

daichan2019 committed:
[Fix] refs #1

headerのnavのactiive状態では線を消すように修正（解決できていない）

daichan2019 committed:
[Fix] refs #1

お知らせ記事ページのタイトルの余計なpaddingを消去

daichan2019 committed:
[Fix] refs #1

footer navからスムーススクロールでリンクへ飛べる機能の実装

daichan2019 commented:
ae1936b
川村さんとのやりとりであったお知らせ一覧ページのタイトルの余白を消しました。

namitarox commented:
これ、min-widthも入ってない

daichan2019 commented:
> これ、min-widthも入ってない

サイト全体的にって認識でしょうか？
最低限レスポンシブに対応できるように修正した方が良いとの意味とくみ取りました
それともある特定の要素のmin-widthを指しているでしょうか？

namitarox commented:
>サイト全体的にって認識でしょうか？
> 最低限レスポンシブに対応できるように修正した方が良いとの意味とくみ取りました
> それともある特定の要素のmin-widthを指しているでしょうか？

ウィンドウ幅狭めたときに、レイアウト崩壊してるから、特定の幅以下になったときは横スクロール表示させる方がいいと思うねんけど。
見た感じ1100px

daichan2019 committed:
[Fix] refs #1

footer navのリンクの修正

daichan2019 committed:
[Fix] refs #1

headerの緑色のボタンだけ背景色が付いているところのスタイリングの見直し

daichan2019 commented:
bca9b71
adfadcc

それぞれ修正しました。
返信の形でコメント残しています。

min-width当てる件に関しては、修正保留にしています。

namitarox commented:
min-width 1100pxでオネシャス

namitarox commented on wp-content/base/scss/layout/_l-first-view.scss:
```diff
@@ -0,0 +1,15 @@
+.l-first-view {
+  background-image: url(/wp-content/base/images/01_top_page/top-page-first-view.jpg);
+  background-position: center -90px;
```
これ、-90pxつけてる理由は？
あと、幅を狭くすると縦に画像のリピートが入るから、xにno-repeat入れる。

no-repeat、幅が狭まるほど下に余白ができるから、-90pxあると調整しづらいかも

daichan2019 replied:
当初、レスポンシブ対応不要の認識だったので、PC版でカンプと同じ高さに背景画像が切り出されるように、微調整していました。（ background-position: center center;）だと、看護師の目がheaderと被ってしまうなど、全てのページにみられたので上からマイナスで微調整していました。

こちらbackground-position: center center;で合わせてしまってよろしいでしょうか？？

namitarox replied:
> こちらbackground-position: center center;で合わせてしまってよろしいでしょうか？？

これでOK
結局-90pxしても、今度は横幅が1100pxになったときにほぼheaderと目が被るからあんま状況変わらんかと。

daichan2019 replied:
8897078

TOPページ含めて、各ページのfirst-view
background-position: center center;
background-repeat: no-repeat;
に変更

namitarox reviewed (commented):
コメントしました！
min-widthと合わせて対応してください

daichan2019 commented:
> min-width 1100pxでオネシャス

承知しました。
何度も確認申し訳ないですが、min-width1100pxで合わせるということは、画面幅1100px以下には縮まらない、という認識でよろしかったでしょうか？？
pc版から作って、多少レスポンシブに合わせるならば、`max-width: 980pxとwidth: 96%;`などで対応しています。

namitarox commented:
>承知しました。
何度も確認申し訳ないですが、min-width1100pxで合わせるということは、画面幅1100px以下には縮まらない、という認識でよろしかったでしょうか？？

その認識でOK
まぁ、余白とって、1120とかでもいいけど任せる。

daichan2019 committed:
[Fix] refs #1

各ページのfirst-view
background-position: center center;
background-repeat: no-repeat;
に変更

daichan2019 committed:
[Fix] refs #1

header, wrapper, footerのmin-widthを1120pxに設定
（ただし、sidebarを含むページは980pxに設定）

daichan2019 commented:
> > 承知しました。
> > 何度も確認申し訳ないですが、min-width1100pxで合わせるということは、画面幅1100px以下には縮まらない、という認識でよろしかったでしょうか？？
> 
> その認識でOK
> まぁ、余白とって、1120とかでもいいけど任せる。

84bc496
header ,wrapper, footerの`min-width:1120px;`にしました。
ただし、sidebarを含むページのwrapperはmin-width:980px;で対応しています。

daichan2019 committed:
[Fix] refs #1

- [x] ➀事業所のご案内：「生協きらり助産院」の列ごと削除→修正

daichan2019 committed:
[Fix] refs #1

➁「医師・医学生」「企画」の「私たちは夢に向かって頑張る医学生をサポートするために、様々な企画をご用意しています。」をフォントサイズ：16pxへ変更→修正

daichan2019 committed:
[Fix] refs #1

- [ ] ➂「医師・医学生」「企画」内の【フィールドワーク】実際の現場に行き、生の声を聞くことができます。長島愛生園にてハンセン病について学習。
→【フィールドワーク】現地に行き、生の声を聞くことができます。長島愛生園にてハンセン病について学習。へ変更

daichan2019 committed:
[Fix] refs #1

「医師・医学生」「企画」内の画像を新たにトリミングされたものに変更

daichan2019 committed:
[Fix] refs #1

➃「医師・医学生」「学生医サポートセンター」中身の紹介の画像テキストの変更
┗画像：Figmaを参考にしてくださいグレーの背景に白の文字、今後画像が入ることを見越して空けておいてください）
┗テキスト：「coming soon」
→画像の差し替えと、テキストを「coming soon」に変更

daichan2019 committed:
[Fix] refs #1

 ⑥「医師・医学生」の「病院見学・実習」
「斐川生協病院」の画像テキスト変更
┗画像：Figmaを参考にしてください（グレーの背景に白の文字、今後画像が入ることを見越して空けておいてください）
┗テキスト：「coming soon」
→画像と画像内のテキスト「coming soon」に変更

daichan2019 committed:
[Fix] refs #1

⑦「医師・医学生」「初期研修・後期研修」の文字サイズの変更
┗サイズ：14pxから16pxへ変更
→修正

daichan2019 committed:
[Add][Fix] refs #1

Google Font Noto Serif JP', serifの追加
各ページのfirst-viewのキャッチコピーのフォントを Noto Serif JP', serifに変更修正

daichan2019 commented:
ee0af32
➀事業所のご案内：「生協きらり助産院」の列ごと削除→修正

daichan2019 commented:
dac3561
➁「医師・医学生」「企画」の「私たちは夢に向かって頑張る医学生をサポートするために、様々な企画をご用意しています。」をフォントサイズ：16pxへ変更
→修正

daichan2019 commented:
86a5eaf
➂「医師・医学生」「企画」内の【フィールドワーク】実際の現場に行き、生の声を聞くことができます。長島愛生園にてハンセン病について学習。
→【フィールドワーク】現地に行き、生の声を聞くことができます。長島愛生園にてハンセン病について学習。へ変更

daichan2019 commented:
3b53017
➃「医師・医学生」「学生医サポートセンター」中身の紹介の画像テキストの変更
┗画像：Figmaを参考にしてくださいグレーの背景に白の文字、今後画像が入ることを見越して空けておいてください）
┗テキスト：「coming soon」
→画像の差し替えと、テキストを「coming soon」に変更

daichan2019 commented:
84997e3

⑥「医師・医学生」の「病院見学・実習」
「斐川生協病院」の画像テキスト変更
┗画像：Figmaを参考にしてください（グレーの背景に白の文字、今後画像が入ることを見越して空けておいてください）
┗テキスト：「coming soon」
→画像と画像内のテキスト「coming soon」に変更

daichan2019 commented:
e6647c2
⑦「医師・医学生」「初期研修・後期研修」の文字サイズの変更
┗サイズ：14pxから16pxへ変更
→修正

daichan2019 commented:
4693868

Google Font Noto Serif JP', serifの追加
各ページのfirst-viewのキャッチコピーのフォントを Noto Serif JP', serifに変更修正

daichan2019 committed:
[Fix] refs #1

「医師を目指す高校生」の「高校生向け企画」内の画像をFigmaでトリミングされたものへ変更

daichan2019 commented:
9178a41

「医師を目指す高校生」の「高校生向け企画」内の画像をFigmaでトリミングされたものへ変更

daichan2019 committed:
[Fix] refs #1

➄「看護学生」「学生医サポートセンター」の中身の画像テキストの変更
┗画像：Figmaを参考にしてください（グレーの背景に白の文字、今後画像が入ることを見越して空けておいてください）
┗テキスト：「coming soon」
→画像とコメントの差し替えをしました

daichan2019 commented:
09d1754

➄「看護学生」「学生医サポートセンター」の中身の画像テキストの変更
┗画像：Figmaを参考にしてください（グレーの背景に白の文字、今後画像が入ることを見越して空けておいてください）
┗テキスト：「coming soon」
→画像の差し替えと、テキストを「coming soon」に変更

daichan2019 commented:
川村さんからLINEのノートでいただいた修正内容、first-viewのキャッチコピーのフォントをgoogle fontに変更

上記共に全て修正完了しました。

daichan2019 committed:
[Fix] refs #1

トップページのLINEのアンカーを設定
アンカー先へスクロールして飛ぶJSの修正

daichan2019 committed:
[Fix] refs #1

各ページのお知らせ部分
画像差し替えとテキストの変更

daichan2019 commented:
7a6af2f
footerのLINEアイコンからsidebarのLINEへアンカーを設定
アンカー先へスクロールして飛ぶJSも追加で修正

daichan2019 commented:
43871ad

各ページのお知らせ部分
画像差し替えとテキストの変更

namitarox reviewed (commented):
LGTM
